File: servers/assessment/assessments/assessmentDTO/assessmentResults.go
package assessmentDTO

import (
	"github.com/google/uuid"
	"github.com/prompt-edu/prompt/servers/assessment/assessments/actionItem/actionItemDTO"
	"github.com/prompt-edu/prompt/servers/assessment/assessments/assessmentCompletion/assessmentCompletionDTO"
	"github.com/prompt-edu/prompt/servers/assessment/assessments/scoreLevel/scoreLevelDTO"
)

// AggregatedEvaluationResult represents anonymized, averaged evaluation results for a competency.
type AggregatedEvaluationResult struct {
	CompetencyID        uuid.UUID `json:"competencyID"`
	AverageScoreNumeric float64   `json:"averageScoreNumeric"`
}

// StudentAssessmentResults bundles all information a student may see once results are released.
type StudentAssessmentResults struct {
	CourseParticipationID uuid.UUID                                    `json:"courseParticipationID"`
	CoursePhaseID         uuid.UUID                                    `json:"coursePhaseID"`
	AssessmentCompletion  assessmentCompletionDTO.AssessmentCompletion `json:"assessmentCompletion"`
	Assessments           []Assessment                                 `json:"assessments"`
	StudentScore          scoreLevelDTO.StudentScore                   `json:"studentScore"`
	PeerEvaluationResults []AggregatedEvaluationResult                 `json:"peerEvaluationResults"`
	SelfEvaluationResults []AggregatedEvaluationResult                 `json:"selfEvaluationResults"`
	ActionItems           []actionItemDTO.ActionItem                   `json:"actionItems,omitempty"`
}

File: clients/assessment_component/src/assessment/interfaces/assessmentResults.ts
import type { ActionItem } from './actionItem'
import type { Assessment } from './assessment'
import type { AssessmentCompletion } from './assessmentCompletion'
import type { CategoryAssessment } from './categoryAssessment'
import type { StudentScore } from './studentScore'

export interface AggregatedEvaluationResult {
  competencyID: string
  averageScoreNumeric: number
}

export interface StudentAssessmentResults {
  courseParticipationID: string
  coursePhaseID: string
  assessmentCompletion: AssessmentCompletion
  assessments: Assessment[]
  categoryAssessments: CategoryAssessment[]
  studentScore?: StudentScore
  peerEvaluationResults: AggregatedEvaluationResult[]
  selfEvaluationResults: AggregatedEvaluationResult[]
  actionItems?: ActionItem[]
}
